src/module.c

```diff
@@ -8569,17 +8571,26 @@ void RM_ThreadSafeContextUnlock(RedisModuleCtx *ctx) {
 }
 
 void moduleAcquireGIL(void) {
-    pthread_mutex_lock(&moduleGIL);
+    pthread_mutex_lock(&moduleGIL.mutex);
+    atomicSet(moduleGIL.owner, pthread_self());
 }
 
 int moduleTryAcquireGIL(void) {
-    return pthread_mutex_trylock(&moduleGIL);
+    int ret = pthread_mutex_trylock(&moduleGIL.mutex);
+    if (ret == 0) atomicSet(moduleGIL.owner, pthread_self());
+    return ret;
 }
 
 void moduleReleaseGIL(void) {
-    pthread_mutex_unlock(&moduleGIL);
+    atomicSet(moduleGIL.owner, (pthread_t)0);
```

Comment:
according to the spec, i don't think we can rely on 0 not being the real ID of some thread.
in practice, i suppose it's not a real concern.